openapi.yml

```diff
@@ -0,0 +1,381 @@
+        '400':
+          $ref: '#/components/responses/Invalid'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          description: Unexpected error
+  /runs:
+    get:
+      summary: Return a list of all runs
+      tags:
+        - runs
+      responses:
+        '200':
+          description: An array of Run objects
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RunArray'
+        '400':
+          $ref: '#/components/responses/Invalid'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          description: Unexpected error
+    post:
+      summary: Creates a run entry
+      tags:
+        - runs
+      security: 
+        - apiKey: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Run'
+      responses:
+        '201':
+          description: created
+        '400':
+          $ref: '#/components/responses/Invalid'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        default:
+          description: Unexpected error
+  /runs/{runId}: 
+    get:
+      summary: Return detailed information about a specific run
+      tags:
+        - runs
+      parameters:
+        - in: path
+          name: runId
+          schema:
+            type: integer
+          required: true
+          description: Run ID of the run to fetch
+      responses:
+        '200':
+          description: A Run object
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Run'
+        '400':
+          $ref: '#/components/responses/Invalid'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          description: Unexpected error
+  /servers:
+    get:
+      summary: Return an array of all connected servers
+      tags:
+        - servers
+      responses:
+        '200':
+          description: An array of Server objects
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ServerArray'
+        '400':
+          $ref: '#/components/responses/Invalid'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          description: Unexpected error
+
+# We can store common response objects and types here
+components:
+  # Security schemes for protected endpoints (run submission)
+  securitySchemes:
+    ApiKeyAuth:
+      type: apiKey
+      in: header
+      name: X-API-Key
```

Comment:
iirc there were discussions of using next-auth, might be better to run w/ JWT in this case

Reply:
atm everything relating to runs /POST is just for reference while the api is written, it'll be removed from docs later, endpoint will require an api key to access from the server plugin (at this point in time) - next-auth would be used for site login stuff i imagine